cypress/integration/scheduled_events_spec.js

```diff
@@ -0,0 +1,187 @@
+      location: 'A remote island of choice',
+    },
+    invitees_counter: {
+      total: 1,
+      active: 1,
+      limit: 1,
+    },
+  },
+  {
+    uri: 'https://api.calendly.com/scheduled_events/GBGBDCAADAEDCRZ4',
+    name: 'Second chat',
+    date: '05/03/2022',
+    start_time: '2022-05-03T19:00:00.000Z',
+    start_time_formatted: '03:00 PM',
+    end_time_formatted: '03:30 PM',
+    status: 'canceled',
+  },
+  {
+    uri: 'https://api.calendly.com/scheduled_events/GFBGBDCAADAEDCRZ4',
+    name: 'Third chat',
+    date: '11/11/2022',
+    start_time: '2022-11-11T20:00:00.000Z',
+    start_time_formatted: '03:00 PM',
+    end_time_formatted: '03:30 PM',
+    status: 'active',
+  },
+];
+
+describe('Scheduled Events', () => {
+  it('Should render a table of all scheduled events, allowing filtering by all, active, and canceled events', () => {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/oauth/authorize*',
+      },
+      (req) => {
+        req.redirect(
+          `${
+            Cypress.config().baseUrl
+          }/oauth/callback?code=5BbtpL2SJIeDP4yClOJPHMJZwEDF1QkbPNaJgkTymeI`,
+          302
+        );
+      }
+    );
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/event_types',
+      },
+      {
+        eventTypes: [],
+      }
+    );
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/scheduled_events*',
+      },
+      {
+        events: scheduledEvents,
+      }
+    );
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/scheduled_events?&status=active',
+      },
+      {
+        events: [scheduledEvents[0], scheduledEvents[2]],
+      }
+    );
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/scheduled_events?&status=canceled',
+      },
+      {
+        events: [scheduledEvents[1]],
+      }
+    );
+
+    cy.visit('/login');
+    cy.get('.btn-large').click();
+    cy.get('nav').contains('Events').click();
+    cy.get('td').eq(0).should('have.text', 'First chat');
+    cy.get('td').eq(1).should('have.text', '04/28/2022');
+    cy.get('td').eq(2).should('have.text', '04:00 PM');
+    cy.get('td').eq(3).should('have.text', '05:00 PM');
+    cy.get('td').eq(4).should('have.text', 'ACTIVE');
+    cy.get('td').eq(5).should('have.text', 'Second chat');
+    cy.get('td').eq(6).should('have.text', '05/03/2022');
+    cy.get('td').eq(7).should('have.text', '03:00 PM');
+    cy.get('td').eq(8).should('have.text', '03:30 PM');
+    cy.get('td').eq(9).should('have.text', 'CANCELED');
+    cy.get('td').eq(10).should('have.text', 'Third chat');
+    cy.get('td').eq(11).should('have.text', '11/11/2022');
+    cy.get('td').eq(12).should('have.text', '03:00 PM');
+    cy.get('td').eq(13).should('have.text', '03:30 PM');
+    cy.get('td').eq(14).should('have.text', 'ACTIVE');
+    cy.get('.css-4xgw5l-IndicatorsContainer2').click();
+    cy.get('.css-1plh46m-MenuList2').contains('Active Events').click();
```

Comment:
Is `.css-1plh46m-MenuList2` always going to be the same? I'm not sure how this works under the hood, but I see that hash, and I worry css changes or recompiling might change that. I'm not sure though

Reply:
That's a great callout. I'm not sure. I remember having problems using the dev-implemented selectors, so had to search around in Chrome dev tools for this cryptic selector. Haha. I may go back to make it more dev-controllable.

Reply:
Yea recompiling/changing css/resetting cypress will probably cause these to fail